fdbserver/CMakeLists.txt

```diff
@@ -21,6 +21,8 @@ endif()
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/workloads)
 
 add_flow_target(EXECUTABLE NAME fdbserver SRCS ${FDBSERVER_SRCS})
+set_target_properties(fdbserver PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/fdbserver.map")
```

Comment:
Can you explain why this is needed?

Reply:
This is a mistake, I thought I could export the dynamic symbols using a linker map but failed and forgot to remove it. `--export-dynamic` is the only option needed on Linux, I will find the equivalent on mac.

Reply:
The only equivalent I found is `-exported_symbols_list` which takes a file in parameter with the list of symbols to export. I'll commit this, but I have no way to test it.

Reply:
I forgot to push.
No problem, thank you as well.

Reply:
Different mac compilation failure now:

```
ld: unknown option: -exported_symbols_list=/Users/.../fdbserver/workloads/external_workload_exports
.txt
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Logs url: https://github.com/apple/foundationdb/pull/11288#issuecomment-2477546008

Reply:
Strange, I'll look tomorrow if I find a solution.
The simpler option for exporting symbols is to compile a shared library (libclient_workload.so for example), but I prefer not to have an additional library to link against.
Exporting the function pointers at runtime would be another solution like the "object-oriented" design, but not sure it's worth complexifying the API for a compilation problem.

Reply:
Or I just formatted it wrong... apparently it's a comma not an equal I had to put between the option and the file path. I pushed again, but not sure at all it will work this time.

Reply:
Triggered CI again.